src/Components/SingleArticleList.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from "react";
+import { fetchData, patchArticle } from "../utils/api";
+import { useParams } from "react-router-dom";
+import CommentsList from "./CommentsList";
+
+function SingleArticleList () {
+    const [article, setArticle] = useState({})
+    const [isLoading, setIsLoading] = useState(true)
+    const [votes, setVotes] = useState(0)
+    const [votesLoading, setVotesLoading] = useState(false)
+    const { article_id } = useParams()
+
+    useEffect(() => {
+        setIsLoading(true)
+        fetchData(`/articles/${article_id}`).then((data) => {
+            setArticle(data.article)
+            setVotes(data.article.votes)
+            setIsLoading(false)
+        }).catch((err) => {
+            console.log(err);
+        })
+    }, [article_id])
+
+    function handleUpvote (article_id) {
+        setVotesLoading(true)
+        setVotes(votes +1)
+        patchArticle(article_id, 1)
+        .then((data) => {
+            setVotes(data.article.votes)
+            setVotesLoading(false)
+            })
+        .catch((err) => {
+            console.log(err);
+            setVotes(votes)
+            setVotesLoading(false)
+        })
+    }
+
+    function handleDownvote (article_id) {
+        setVotesLoading(true)
+        setVotes(votes -1)
+        patchArticle(article_id, -1)
+        .then((data) => {
+            setVotes(data.article.votes)
+            setVotesLoading(false)
+            })
+        .catch((err) => {
+            console.log(err);
+            setVotes(votes)
+            setVotesLoading(false)
+        })
+    }
+
+    return isLoading ? (
+        <h2>Loading...</h2>
+    ) : (
+        <div className='article'>
+            {article.title && <h3>{article.title}</h3>}
+            {article.author && <p>By {article.author}</p>}
+            <p>Votes: {votesLoading ? '...' : votes}</p>
```

Comment:
This way to deal with votes is alright, but it would be better to do some optimistic rendering and set the votes back afterwards if an error occurs